From akpm@linux-foundation.org Thu Apr 26 00:38:09 2007
From: "Wu, Bryan" <bryan.wu@analog.com>
Date: Thu, 26 Apr 2007 00:38:01 -0700
Subject: USB gadget rndis: fix struct rndis_packet_msg_type unaligned bug
To: greg@kroah.com
Cc: linux-usb-devel@lists.sourceforge.net, akpm@linux-foundation.org, bryan.wu@analog.com, david-b@pacbell.net, jie.zhang@analog.com, roy.huang@analog.com
Message-ID: <200704260738.l3Q7c22A024368@shell0.pdx.osdl.net>


From: "Wu, Bryan" <bryan.wu@analog.com>

skb_push function may return a pointer which is not aligned as required
by struct rndis_packet_msg_type. Using attribute trick to fix this bug.

Signed-off-by: Roy Huang <roy.huang@analog.com>
Signed-off-by: Jie Zhang <jie.zhang@analog.com>
Signed-off-by: Bryan Wu <bryan.wu@analog.com>
Cc: David Brownell <david-b@pacbell.net>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/rndis.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/gadget/rndis.h
+++ b/drivers/usb/gadget/rndis.h
@@ -195,7 +195,7 @@ struct rndis_packet_msg_type
 	__le32	PerPacketInfoLength;
 	__le32	VcHandle;
 	__le32	Reserved;
-};
+} __attribute__ ((packed));
 
 struct rndis_config_parameter
 {
